metals/src/main/scala/scala/meta/internal/metals/config/RunType.scala

```diff
@@ -3,12 +3,14 @@ package scala.meta.internal.metals.config
 object RunType {
   sealed trait RunType
   case object Run extends RunType
+  case object RunFile extends RunType
   case object TestFile extends RunType
   case object TestTarget extends RunType
 
   def fromString(string: String): Option[RunType] = {
     string match {
       case "run" => Some(Run)
+      case "runFile" => Some(RunFile)
```

Comment:
🤔 the only thing that might be confusing about this is that depending on how the client sets it, it might totally not be clear. For example `nvim-metals` users will have this in their configuration if they use dap:

```
  dap.configurations.scala = {
    {
      type = "scala",
      request = "launch",
      name = "Run",
      metals = {
        runType = "run",
        --args = { "firstArg", "secondArg", "thirdArg" }, -- here just as an example
      },
    },
    {
      type = "scala",
      request = "launch",
      name = "Test File",
      metals = {
        runType = "testFile",
      },
    },
    {
      type = "scala",
      request = "launch",
      name = "Test Target",
      metals = {
        runType = "testTarget",
      },
    },
  }
```
So we'd just need to add another one that sends in `runFile`, but it's not really clear what the difference between `run` and `runFile` is without knowing how these work. Do you think maybe just having these documented super well is enough, or can you think of another name that would maybe be clearer? I'm sort of failing on coming up with anything better, so maybe it's fine. idn. wdyt?

Reply:
`runOrTestFile` ? Or maybe `invokeFile`? Or just `debugFile` ? I wasn't perfectly sure about the name either.

Reply:
Thinking about this a bit, maybe we went about this a bit wrong when we first did it. Maybe instead of testFile, testTarget, run, etc we could simplify it.

`runOrTestFile` -> executes the main or the tests in the file
`runOrTestTarget` -> runs the main or tests in the build target

Actually now that I'm thinking about it, this may not work in the scenario you have a main in your build target. Or maybe it would? We could detect that their are both and just ask the user if they want to run or test? Not sure.

I'd say if we keep the current way then maybe be explicit and just change it to `runOrTestFile`, but I don't feel strong about it.

if the other option would work maybe it'd be nicer for the user? Really not sure.

Reply:
I think we could add `runOrTestFile` and then deprecate `run` and `testFile`?

I am not so sure about `runOrTestTarget` - I think `testTarget` is ok, since it's a different logic, user wants to run all tests and not just choose a test/main, so let's not add an additional step there. We could have a fallback for `runOrTestFile` that tries main or tests from other files, but it could be ok to not include it for now. `runOrTestTarget` would need to have to different purposes - run a single main or run all tests, which is kind of weird.

I think all of those should server a UX purpose:
- `runOrTestFile` - run whatever I am working on quickly
- `testTarget` - let's run all tests to see if I didn't break anything

Reply:
Renamed it to `runOrTestFile`, let me know what you think!